roles/libvirt_manager/tasks/prepare_image.yml

```diff
@@ -0,0 +1,139 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Prepares the virtual machine boot image. Though the generation is
+# straight forward, we need to ensure support exists for static IP address
+# configuration.
+#
+# Multiple images are created for a single vm type in the case of static IP
+# address. Also, the index is used to determine the next usable IP address.
+
+- name: Prepare the ecosystem to support static IP addressing.
+  when:
+    - vm_data.value.ip_address is defined
+  block:
+    - name: Generate the static address configuration script
+      vars:
+        _ip_address: >-
+          {{
+            vm_data.value.ip_address.address |
+            ansible.utils.next_nth_usable(vm_id)
+          }}
+        _ip_gw: "{{ vm_data.value.ip_address.gw }}"
+        _ip_dns: "{{ vm_data.value.ip_address.dns }}"
+        _ip_prefix: >-
+          {{
+            vm_data.value.ip_address.address | ansible.utils.ipaddr('prefix')
+          }}
+        _file_suffix: >-
+          {{
+            (vm_data.value.amount is defined) |
+            ternary(
+              'static-ip-' ~ vm_data.key ~ '-' ~ vm_id ~ '.sh',
+              'static-ip-' ~ vm_data.key ~ '.sh'
+            )
+          }}
+      ansible.builtin.template:
+        src: templates/static_ip.j2
+        dest: >-
+          {{
+            (cifmw_libvirt_manager_ip_script_basedir, _file_suffix) |
+            ansible.builtin.path_join
+          }}
+        owner: "{{ ansible_user_id }}"
+        group: "{{ ansible_user_id }}"
+        mode: "0755"
+      loop: "{{ range(0, vm_data.value.amount | default(1) | int) }}"
+      loop_control:
+        index_var: vm_id
+        label: "{{ vm_data.key }}-{{ vm_id }} "
+
+    - name: Ensure separate images exists for each VM
```

Comment:
why copy? Why don't you just inject the "specific" content in the overlay instead? It takes far, far less space using overlays... I don't get it.

Reply:
Done - Moved the modify the overlay instead.